litedram/phy/s7ddrphy.py

```diff
@@ -50,9 +50,9 @@ def __init__(self, pads, with_odelay,
         bankbits    = len(pads.ba) if memtype != "DDR4" else len(pads.ba) + len(pads.bg)
         nranks      = 1 if not hasattr(pads, "cs_n") else len(pads.cs_n)
         databits    = len(pads.dq)
+        strobes     = len(pads.dqs_p)
         nphases     = nphases
         assert databits%8 == 0
-        x4_dimm_mode = (databits / len(pads.dqs_p)) == 4
```

Comment:
we may have an assertion here that verifies that the `DQ/DQS` ratio is either `4` or `8`

Reply:
Probably we can add it when the dq dqs ratio is defined below

Reply:
Done